src/common/utility.ts

```diff
@@ -106,4 +182,32 @@ export class Utility {
             setTimeout(resolve, ms);
         });
     }
+
+    private static async pickConnection(context: vscode.ExtensionContext): Promise<IConnection> {
+        let selectedConnection;
+
+        const connections = context.globalState.get<{ [key: string]: IConnection }>(Constants.GlobalStateMySQLConectionsKey);
+        const items: vscode.QuickPickItem[] = [];
+
+        if (connections) {
+            for (const id of Object.keys(connections)) {
+                const item = connections[id];
+                items.push({
+                  label: item.host,
+                  description: item.user});
+            }
+        }
+
+        await vscode.window.showQuickPick(items).then(async (selection) => {
+            // the user canceled the selection
+            if (!selection) {
+                return selectedConnection;
+            }
+
+            const selectedConnectionId = Object.keys(connections)[items.indexOf(selection)];

```

Comment:
What if there are two connections have the same host and user?

Reply:
You would still get the correct selection from the pick menu.. but you wont be able to differ from the two when the menu is shown.. how would you enable anyone to see the difference between two connections with the same host and user? showing the internal id of the connection wont be to much use for the user.

Reply:
If we could still get the correct selection from the pick menu, then it will be OK. No extra change needed.